File: Weather/app/src/main/java/com/devaman/weather/javaFile/MainActivity.java
package com.devaman.weather.javaFile;

import androidx.appcompat.app.AppCompatActivity;

import java.util.Date;
import java.text.SimpleDateFormat;
import android.os.Bundle;
import android.view.View;
import android.view.textclassifier.ConversationActions;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;
import com.devaman.weather.R;

import org.json.JSONException;
import org.json.JSONObject;

public class MainActivity extends AppCompatActivity {

    View view;
    EditText editTextSearch;
    TextView temp_tv, city_tv, desc_tv, humidity_tv, min_temp_tv, max_temp_tv, wind_tv, sunrise_tv, visibility_tv, sunset_tv, pressure_tv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editTextSearch = (EditText) findViewById(R.id.editTextSearch);
        temp_tv = (TextView) findViewById(R.id.temp_tv);
        min_temp_tv = (TextView) findViewById(R.id.min_temp_tv);
        max_temp_tv = (TextView) findViewById(R.id.max_temp_tv);

	    city_tv = (TextView) findViewById(R.id.city_tv);
        desc_tv = (TextView) findViewById(R.id.desc_tv);
        humidity_tv = (TextView) findViewById(R.id.humidity_tv);
        pressure_tv = (TextView) findViewById(R.id.pressure_tv);

        wind_tv = (TextView) findViewById(R.id.wind_tv);
        sunrise_tv = (TextView) findViewById(R.id.sunrise_tv);
        visibility_tv = (TextView) findViewById(R.id.visibility_tv);
        sunset_tv = (TextView) findViewById(R.id.sunset_tv);

        editTextSearch.setText("New Zealand");
        getWeather(getCurrentFocus());
    }
    public void getWeather(View view){
        String apikey = "d4c70ae853a0c4b4a963a1c4b720234d";
        String city = editTextSearch.getText().toString();
        editTextSearch.setText("");
        String url_byCity = "https://api.openweathermap.org/data/2.5/weather?q="+city+"&appid="+apikey;

        RequestQueue requestQueue = Volley.newRequestQueue(getApplicationContext());
        JsonObjectRequest request = new JsonObjectRequest(Request.Method.GET, url_byCity, null, new Response.Listener<JSONObject>() {
            @Override
            public void onResponse(JSONObject response) {
                try {
                    String str;

                    // Temperature
                    JSONObject mainObject = response.getJSONObject("main");
                    str = mainObject.getString("temp");
                    str = String.valueOf((int) (Float.parseFloat(str)-273.15f) + "°C");
                    temp_tv.setText(str);

                    str = mainObject.getString("temp_min");
                    str = String.valueOf((int) (Float.parseFloat(str)-273.15) + "°C");
                    min_temp_tv.setText(str);

                    str = mainObject.getString("temp_max");
                    str = String.valueOf((int) Math.ceil(Float.parseFloat(str)-273.15) + "°C");
                    max_temp_tv.setText(str);

                    city_tv.setText(response.getString("name"));

                    desc_tv.setText(response.getJSONArray("weather").getJSONObject(0).getString("description"));

                    str = response.getJSONObject("wind").getString("speed");
                    str = String.valueOf((int) (Float.parseFloat(str)*3.6) + "km/hr");
                    wind_tv.setText(str);

                    pressure_tv.setText(response.getJSONObject("main").getString("pressure") + "hPa");

                    humidity_tv.setText(response.getJSONObject("main").getString("humidity")+"%");


                    long unixTimestamp = Long.parseLong(response.getJSONObject("sys").getString("sunrise")); // Example sunrise timestamp
                    Date sunriseDate = new Date(unixTimestamp * 1000); // Convert to milliseconds
                    //System.out.println(sunriseDate); // Output: Wed May 19 05:43:42 GMT 2021
                    // Format the date and time in 24-hour format
                    SimpleDateFormat sdf = new SimpleDateFormat("HH:mm");
                    String formattedSunrise = sdf.format(sunriseDate);
                    sunrise_tv.setText(formattedSunrise);


                    unixTimestamp = Long.parseLong(response.getJSONObject("sys").getString("sunset")); // Example sunrise timestamp
                    Date sunsetDate = new Date(unixTimestamp * 1000); // Convert to milliseconds
                    //System.out.println(sunsetDate); // Output: Wed May 19 05:43:42 GMT 2021
                    // Format the date and time in 24-hour format
                    SimpleDateFormat ssdf = new SimpleDateFormat("HH:mm");
                    String formattedSunset = ssdf.format(sunsetDate);
                    sunset_tv.setText(formattedSunset);


                    str = response.getString("visibility");
                    str = String.valueOf((int) (Float.parseFloat(str)/1000) + "km");
                    visibility_tv.setText(str);

                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                Toast.makeText(getApplicationContext(), error.toString(), Toast.LENGTH_LONG).show();
            }
        });
        requestQueue.add(request);

    }
}

File: Weather/app/src/main/java/com/devaman/weather/MainActivity.kt
package com.devaman.weather

import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.EditText
import android.widget.ImageButton
import android.widget.TextView
import android.widget.Toast
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.Volley
import org.json.JSONObject
import org.w3c.dom.Text
import java.lang.Math.ceil

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val lat = intent.getStringExtra("lat")
        val long = intent.getStringExtra("long")
        Toast.makeText(this, "Latitude : "+lat+"\nLongitude : "+long,Toast.LENGTH_LONG).show()

        // Change status bar Color
        window.statusBarColor= Color.parseColor("#1318c3")

        val apikey = "69b247caf0deb9022639eb57f51e9961"
        val simpleurl = "https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${long}&appid=${apikey}"
        // Get Data from API
        getJsonData(lat, long, simpleurl)

        val buttonSearch = findViewById<ImageButton>(R.id.buttonSearch) as ImageButton
        buttonSearch.setOnClickListener{
            val editTextSearch = findViewById<EditText>(R.id.editTextSearch).text.toString()
            val url_byCity = "https://api.openweathermap.org/data/2.5/weather?q=${editTextSearch}&appid=69b247caf0deb9022639eb57f51e9961"
            getJsonData(lat, long, url_byCity)
        }
    }

    private fun getJsonData(lat: String?, long: String?, url:String) {

         // Instantiate the RequestQueue.

        val queue = Volley.newRequestQueue(this)


        // Request a string response from the provided URL.
        val jsonRequest = JsonObjectRequest(
            Request.Method.GET, url, null,
            Response.Listener { response ->
                // Toast.makeText(this, response.toString(), Toast.LENGTH_LONG).show()
                setValues(response)
        },
            Response.ErrorListener { Toast.makeText(this, "ERROR", Toast.LENGTH_LONG).show()
            })

        // Add the request to the RequestQueue.
        queue.add(jsonRequest)
    }

    private fun setValues(response: JSONObject){
        findViewById<TextView>(R.id.city).text=response.getString("name")
        var lat = response.getJSONObject("coord").getString("lat")
        var long = response.getJSONObject("coord").getString("lon")
        findViewById<TextView>(R.id.coordinates).text="${lat}, ${long}"
        findViewById<TextView>(R.id.weather).text=response.getJSONArray("weather").getJSONObject(0).getString("main")

        var tempr = response.getJSONObject("main").getString("temp")
        tempr=((tempr).toFloat()-273.15).toInt().toString()
        findViewById<TextView>(R.id.temp).text = "${tempr}°C"

        var mintemp = response.getJSONObject("main").getString("temp_min")
        mintemp = (((mintemp).toFloat()-273.15).toInt()).toString()
        findViewById<TextView>(R.id.min_temp).text="${mintemp}°C"

        var maxtemp = response.getJSONObject("main").getString("temp_min")
        maxtemp = (ceil((maxtemp).toFloat()-273.15).toInt()).toString()
        findViewById<TextView>(R.id.max_temp).text="${maxtemp}°C"

        findViewById<TextView>(R.id.pressure).text=response.getJSONObject("main").getString("humidity")+"%"

        findViewById<TextView>(R.id.wind).text=response.getJSONObject("wind").getString("speed")

        findViewById<TextView>(R.id.degree).text=response.getJSONObject("wind").getString("deg")+"°"
    }
}